lib/rucio/common/types.py

```diff
@@ -99,3 +99,43 @@ class InternalScope(InternalType):
     '''
     def __init__(self, scope, vo='def', fromExternal=True):
         super(InternalScope, self).__init__(value=scope, vo=vo, fromExternal=fromExternal)
+
+
+class RSEDomainLANDict(TypedDict):
+    read: int
+    write: int
+    delete: int
+
+
+class RSEDomainWANDict(TypedDict):
+    read: int
+    write: int
+    delete: int
+    third_party_copy_read: int
+    third_party_copy_write: int
+
+
+class RSEDomainsDict(TypedDict):
+    lan: dict[str, RSEDomainLANDict]
+    wan: dict[str, RSEDomainWANDict]
```

Comment:
Not exactly. `lan: dict[str, RSEDomainLANDict]` means the value must resemble:
```json
{
    "foo": {
        "read": 0,
        "write": 0,
        "delete": 0
    }
}
```
to be valid. You want `lan: RSEDomainLANDict` instead (similarly for many others).